org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/extension/AbstractContributionComposite.java

```diff
@@ -101,8 +111,11 @@ public <T> List<T> getContributionsByClass(Class<T> type) {
       throw new IllegalArgumentException("Contribution type class must be specified.");
     }
 
-    ArrayList<T> contributionsOfType = (ArrayList<T>) m_contributionsByType.get(type);
-    if (contributionsOfType == null) {
+    ArrayList<T> contributionsOfType = null;
+    if (m_contributionsByType != null) {
+      contributionsOfType = (ArrayList<T>) m_contributionsByType.get(type);
+    }
+    if (contributionsOfType == null && m_contributionsByClass != null) {
```

Comment:
If performance suggestion is applied:
```suggestion
    if (contributionsOfType == null) {
```